### DISCUSSION TITLE
consider replacing rocksdb with tikv/agatedb

### DISCUSSION THREAD

**Discussion by emberian** [celo-org/optics-monorepo#591] [Ideas]
https://github.com/tikv/agatedb  probably builds faster (i sat there and watched librocksdb-sys block the entire build for a bit)

open questions:
- what rocksdb features do we use?
- is agatedb stable enough?

**Comment by prestwich** [comment]
> what rocksdb features do we use?

We do use iteration over lexically-sorted keys
